### PR TITLE
DataFrame: Deprecate MutableDataFrame

### PR DESCRIPTION
**Pull request by ryantxu**
See https://github.com/grafana/grafana/issues/66480

Now that all DataFrames are "mutable", most things should not need to use MutableDataFrame.  Most of our usage is to support creating a frame from partial input -- it does not usually need the dynamic field registration and clunky mutation helpers

### CODE REVIEW

File: packages/grafana-data/src/dataframe/processDataFrame.ts [hunk 1]
```diff
@@ -567,3 +566,34 @@ export function preProcessPanelData(data: PanelData, lastResult?: PanelData): Pa
     timings: { dataProcessingTime: STOPTIME - STARTTIME },
   };
 }
+
+export interface ParitalDataFrame extends Omit<DataFrame, 'fields' | 'length'> {
```
Comment (leeoniya) [inline packages/grafana-data/src/dataframe/processDataFrame.ts:570]:
```suggestion
export interface PartialDataFrame extends Omit<DataFrame, 'fields' | 'length'> {
```

File: packages/grafana-data/src/dataframe/MutableDataFrame.ts [hunk 2]
```diff
@@ -10,10 +10,16 @@ import { guessFieldTypeFromValue, guessFieldTypeForField, toDataFrameDTO } from
...
 
+/** @deprecated */
 type MutableVectorCreator = (buffer?: any[]) => any[];
 
 export const MISSING_VALUE = undefined; // Treated as connected in new graph panel
 
+/**
+ * MutableDataFrame is a complex wrapper around the DataFrame interface
+ *
+ * @deprecated use standard DataFrame, or create one with ParitalDataFrame
```
Comment (leeoniya) [inline packages/grafana-data/src/dataframe/MutableDataFrame.ts:21]:
```suggestion
 * @deprecated use standard DataFrame, or create one with PartialDataFrame
```

File: packages/grafana-data/src/dataframe/processDataFrame.ts [hunk 3]
```diff
@@ -567,3 +566,34 @@ export function preProcessPanelData(data: PanelData, lastResult?: PanelData): Pa
...
+  fields: Array<Partial<Field>>;
+}
+
+export function createDataFrame(input: ParitalDataFrame): DataFrame {
```
Comment (leeoniya) [inline packages/grafana-data/src/dataframe/processDataFrame.ts:574]:
```suggestion
export function createDataFrame(input: PartialDataFrame): DataFrame {

```

**Review by leeoniya — APPROVED** [review]
LGTM after name fix. let's wait to land this after https://github.com/grafana/grafana/pull/67570 due to merge conflicts and probable backport woes.

File: packages/grafana-data/src/dataframe/processDataFrame.ts [hunk 4]
```diff
@@ -567,3 +566,35 @@ export function preProcessPanelData(data: PanelData, lastResult?: PanelData): Pa
     timings: { dataProcessingTime: STOPTIME - STARTTIME },
   };
 }
+
+export interface PartialDataFrame extends Omit<DataFrame, 'fields' | 'length'> {
+  fields: Array<Partial<Field>>;
+}
+
+export function createDataFrame(input: PartialDataFrame): DataFrame {
```
Comment (ryantxu) [inline packages/grafana-data/src/dataframe/processDataFrame.ts:574]: @leeoniya thoughts on a better/different name?  we have `toDataFrame(any)`

**Pull request merged**